lib/ansible/modules/files/archive.py

```diff
@@ -251,7 +267,7 @@ def main():
     # No source files were found but the named archive exists: are we 'compress' or 'archive' now?
     if len(missing) == len(expanded_paths) and dest and os.path.exists(dest):
         # Just check the filename to know if it's an archive or simple compressed file
-        if re.search(r'(\.tar|\.tar\.gz|\.tgz|.tbz2|\.tar\.bz2|\.zip)$', os.path.basename(dest), re.IGNORECASE):
+        if re.search(r'(\.tar|\.tar\.gz|\.tgz|.tbz2|\.tar\.bz2|\.tar\.xz|\.zip)$', os.path.basename(dest), re.IGNORECASE):
```

Comment:
I just filed https://github.com/ansible/ansible/issues/34119 which, since you're already making changes to this line, you could address.

Reply:
Just add a backslash before `.tbz2`! easy peasy

Reply:
done!